src/components/RegisterForm/RegisterForm.js

```diff
@@ -0,0 +1,129 @@
+import React, { Component } from 'react';
+import ls from 'local-storage';
+import { Link } from 'react-router-dom';
+
+import { Button } from '@/shared';
+
+import './RegisterForm.scss';
+
+const CN = 'register-form';
+
+class RegisterForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: ''
+      // password: '',
+      // name: ''
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    this.setState({
+      email: ls.get('emailToRegister') || ''
+    });
+  }
+
+  saveToLocalStore() {
+
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const form = e.target;
+    const data = new FormData(form);
+
+    fetch('/api/form-submit-url', {
+      method: 'POST',
+      body: data
+    });
+  }
+
+  render() {
+    const { email } = this.state;
+
+    return (
+      <div className={`${CN}`} onSubmit={this.handleSubmit}>
```

Comment:
It is a bad practice to set onSubmit on non-interactive elements (like div). I suggest to change the HTML structure to this one:
form onSubmit={this.handleSubmit}
   div Personal info /div
   div Shipping info /div
/form